lib/controllers/Controller.js

```diff
@@ -54,6 +58,28 @@ Controller.prototype.handleRequest = function (request, response, next, settings
   }
 };
 
+// Get host and protocol from HTTP's Forwarded header
+Controller.prototype._getForwarded = function (request) {
+  if (!request.headers.forwarded)
+    return {};
+  try {
+    var forwarded = _.defaults.apply(this, parseForwarded(request.headers.forwarded));
```

Comment:
Why is this `_.defaults.apply()` needed here?

Reply:
Ah! It's because parseForwarded return a list of objects containing all the host/proto/for/by things it finds and `_.defaults` takes the object list from the arguments. Since we mostly need the first object of the list I thought `[0]` would be good but actually if you have multiple headers and only the first object has proto and the second has host, you actually needs the left-most value for a specific key.

Reply:
I see, but is there any reason to `apply` it to `this`?

Can't you just call `_.defaults(parseForwarded(request.headers.forwarded));`?

Reply:
No it doesn't seem to accept this syntax: https://lodash.com/docs/4.17.4#defaults which makes the test fails.

Reply:
Ah, you're right! I thought it'd accept a single array as well.

I would have implemented is as `_.defaults.apply(_, parseForwarded(request.headers.forwarded));` then, but it doesn't matter in this case, as `_.defaults` is stateless anyways.

Reply:
Yes actually I put null first then wasn't sure... I'm no JS expert but I understand it doesn't use this anyway. I can change to `_` if you think it's cleaner, it doesn't harm.

Reply:
It's fine for me as it is now. Maybe @RubenVerborgh has a preference for one or the other.

Reply:
Fine with me!